netbox/extras/events.py

```
@@ -0,0 +1,146 @@
+    app_label = instance._meta.app_label
+    model_name = instance._meta.model_name
+    if model_name not in registry['model_features']['webhooks'].get(app_label, []):
+        return
+
+    queue.append({
+        'content_type': ContentType.objects.get_for_model(instance),
+        'object_id': instance.pk,
+        'event': action,
+        'data': serialize_for_event(instance),
+        'snapshots': get_snapshots(instance, action),
+        'username': user.username,
+        'request_id': request_id
+    })
+
+
+def process_event_rules(queue):
+    """
+    Flush a list of object representation to RQ for webhook processing.
+    """
+    rq_queue_name = get_config().QUEUE_MAPPINGS.get('webhook', RQ_QUEUE_DEFAULT)
+    rq_queue = get_queue(rq_queue_name)
+    events_cache = {
+        'type_create': {},
+        'type_update': {},
+        'type_delete': {},
+    }
+
+    for data in queue:
+
+        action_flag = {
+            ObjectChangeActionChoices.ACTION_CREATE: 'type_create',
+            ObjectChangeActionChoices.ACTION_UPDATE: 'type_update',
+            ObjectChangeActionChoices.ACTION_DELETE: 'type_delete',
+        }[data['event']]
+        content_type = data['content_type']
+
+        # Cache applicable Event Rules
+        if content_type not in events_cache[action_flag]:
+            events_cache[action_flag][content_type] = EventRule.objects.filter(
+                **{action_flag: True},
+                content_types=content_type,
+                enabled=True
+            )
+        event_rules = events_cache[action_flag][content_type]
+
+        for event_rule in event_rules:
+            if event_rule.action_type == EventRuleActionChoices.WEBHOOK:
+                processor = "extras.webhooks_worker.process_webhook"
+            elif event_rule.action_type == EventRuleActionChoices.SCRIPT:
+                processor = "extras.scripts_worker.process_script"
+            else:
+                return
```

This should probably raise an exception, as it indicates invalid data.